src/OpenLoco/Paint/PaintEntity.cpp

```diff
@@ -36,7 +37,7 @@ namespace OpenLoco::Paint
     }
 
     // 0x0046FA88
-    void paintEntities(PaintSession& session, const Map::map_pos& loc)
+    static void paintEntitiesWithFilter(PaintSession& session, const Map::map_pos& loc, std::function<bool(const thing_base*)> filter)
```

Comment:
Should/can the rtype for `filter` be marked const as well? My thinking is this might enable additional optimisations.